src/Sentry/Http/HttpTransportBase.cs

```diff
@@ -386,9 +386,18 @@ private void HandleFailure(HttpResponseMessage response, Envelope envelope)
                 var destination = Path.Combine(destinationDirectory, "envelope_too_large",
                     (eventId ?? SentryId.Create()).ToString());
 
-                Directory.CreateDirectory(Path.GetDirectoryName(destination)!);
+                if (_options.FileSystem.CreateDirectory(Path.GetDirectoryName(destination)!) is not FileOperationResult.Success)
+                {
+                    _options.LogError("Failed to create directory to store the envelope.");
+                    return;
+                }
```

Comment:
```suggestion
                var createDirectoryResult = _options.FileSystem.CreateDirectory(Path.GetDirectoryName(destination)!);
                if (createDirectoryResult != FileOperationResult.Success)
                {
                    _options.LogError($"Failed to create directory to store the envelope: {createDirectoryResult}");
                    return;
                }
```

Just so you can easily tell in the logs whether it failed or was disabled.